sisl/viz/plotly/plotutils.py

```diff
@@ -672,14 +672,18 @@ def run_multiple(func, *args, argsList = None, kwargsList = None, messageFn = No
     """
 
     #Prepare the arguments to be passed to the initSinglePlot function
+    # TODO could itertools.zip_longest perhaps be used instead?
+    # it sure smells like it?
     toZip = [*args, argsList, kwargsList]
     for i, arg in enumerate(toZip):
         if not isinstance(arg, (list, tuple, np.ndarray)):
             toZip[i] = itertools.repeat(arg)
         else:
+            # TODO also, shouldn't this be max(nTasks, len(arg)) ?
```

Comment:
Hmm actually they all should match and if not when zipping the shortest length will be taken. But this doesn't care about different lengths either so it is wrong as well.

Reply:
ok.

Reply:
A minor question, would it ever happen that *all* arguments in `toZip` gets `itertools.repeat`, because that would be really bad!! ;)

Reply:
Actually yes because the defaults for `argsList` and `kwargsList` are `None`. :scream: 

But as I said in the other comment, this part needs to be rewritten :)

Reply:
Ok, I'll leave it here and once merged you can have a go! :)